chrome/content/helpers.js

```
@@ -0,0 +1,76 @@
+var isJpm = !isNode && (typeof require === "function");
+
+var OS;
+
+if (!isNode) {
+  var Ci, Cu, Cc;
+  if (isJpm) {
+    ({ Ci, Cu, Cc } = require("chrome"));
+  } else {
+    ({ interfaces: Ci, utils: Cu, classes: Cc } = Components);
+    Cu.import("resource:///modules/imServices.jsm");
+    Cu.import("resource:///modules/imXPCOMUtils.jsm");
+  }
+  Cu.import("resource://gre/modules/NetUtil.jsm");
+  Cu.import("resource://gre/modules/FileUtils.jsm");
+  Cu.import("resource://gre/modules/osfile.jsm");
+}
+
+var helpers = {
+  profilePath: function(filename) {
+    return isNode ?
+      path.resolve(__dirname, filename) :
+      OS.Path.join(OS.Constants.Path.profileDir, filename);
+  },
+
+  getAccounts: function* () {
+    let accounts = Services.accounts.getAccounts();
+    while (accounts.hasMoreElements()) {
+      yield accounts.getNext();
+    }
+  },
+
+  readTextFile: function(filename) {
+    let file = new FileUtils.File(filename);
+    if (!file.exists()) {
+      return null;
+    }
+    let data = "";
+    let fstream = Cc["@mozilla.org/network/file-input-stream;1"].
+            createInstance(Ci.nsIFileInputStream);
+    let cstream = Cc["@mozilla.org/intl/converter-input-stream;1"].
+            createInstance(Ci.nsIConverterInputStream);
+    fstream.init(file, -1, 0, 0);
+    cstream.init(fstream, "UTF-8", 0, 0);
+    let str = {};
+    let read = 0;
+    do {
+      read = cstream.readString(0xffffffff, str);
+      data += str.value;
+    } while (read !== 0);
+    cstream.close();
+    return data;
+  },
+
+  writeTextFileAsync: function(data, filename) {
+    let file = new FileUtils.File(filename);
+    let ostream = FileUtils.openSafeFileOutputStream(file);
+    let converter = Cc["@mozilla.org/intl/scriptableunicodeconverter"].
+                    createInstance(Ci.nsIScriptableUnicodeConverter);
+    converter.charset = "UTF-8";
+    let istream = converter.convertToInputStream(data);
+    NetUtil.asyncCopy(istream, ostream);
```

Seem to be ignoring an error here